src/xml2_node.cpp

```diff
@@ -82,15 +170,66 @@ const xmlChar* xmlNsDefinition(xmlNodePtr node, const xmlChar* lookup) {
   return NULL;
 }
 
+
+SEXP node_attr_impl(SEXP x,
+                    std::string name,
```

Comment:
```suggestion
                    const std::string& name,
```

Just to avoid copying in the loop